Include a new downsampling operation (#1574)
elastic/rally: docs/track.rst

* ``weight``: The number of fetched pages. Should always equal to the ``pages`` parameter.
* ``unit``: The unit in which to interpret ``weight``. Always "ops".

> downsampling

Reviewer: What about `downsample`? Similarly to how we call our query operation `search` as opposed to `searching`

> ~~~~~~~~~~~~~~~~~~~

Reviewer: nit
```suggestion
~~~~~~~~~~
```

Executes a downsampling operation on an index producing a new index whose data is aggregated on the @timestamp field.

Properties
""""""""""

* ``fixed_interval``: The aggregation interval key defined as in `https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html#fixed_intervals`.

> * ``source-index``: The read-only index containing data to aggregate which includes a ``@timestamp`` field.

Reviewer: ```suggestion
* ``source-index``: The index containing data to aggregate which includes a ``@timestamp`` field. Note that this index should be marked read-only prior to the execution of this operation.
```

> * ``target-index``: Tne new target index created by the downsamplig operation and including aggregated data.

Reviewer: ```suggestion
* ``target-index``: Tne new target index created by the downsampling operation and including aggregated data.
```

> * ``source-index`` (optional, defaults to ``tsdb-index``): The index containing data to aggregate which includes a ``@timestamp`` field. Note that this index should be marked read-only prior to the execution of this operation.

Reviewer: ```suggestion
``source-index`` (optional): The index containing data to aggregate which includes a ``@timestamp`` field. Note that this index should be marked read-only prior to the execution of this operation. If there is only one index defined in the ``indices`` of the track definition, that will be used as the default.
```

> * ``body`` (mandatory): The ``body`` property must contain the ``fixed_interval`` key defined as in `https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html#fixed_intervals` the ``source-index`` and the ``target-index``. The source index is an existing index which includes a ``@timestamp`` field. Write operations to the source index must be disabled. Tne target index is a new index that is created by the downsamplig operation and including aggregated data.

Reviewer: Most of the time we abstract common or critical `body` elements for the user to explicit parameters. If the `body` has required keys, it's likely better to just parameterize the operation instead of providing the body verbatim. This also gives the operation advantage over just using a `raw-request`.

Author: I was thinking about that and at the end I chose to have the body because I had the feeling that having the body was "more consistent" with other operations. But I see your point here and I will get rid of it.
